cmake/test/nosetests.cmake

```diff
@@ -73,7 +73,15 @@ function(catkin_add_nosetests path)
   set(output_path ${CATKIN_TEST_RESULTS_DIR}/${PROJECT_NAME})
   # make --xunit-file argument an absolute path (https://github.com/nose-devs/nose/issues/779)
   get_filename_component(output_path "${output_path}" ABSOLUTE)
-  set(cmd "${CMAKE_COMMAND} -E make_directory ${output_path}")
+
+  if(WIN32)
```

Comment:
Why should this be Windows specific? Couldn't the same happen on other platforms?

Reply:
Thanks for the question! Yes, supposedly this should be a problem across platforms; we ran into this problem when CMake was installed into a path with spaces. Since the entire command would be passed into https://github.com/ros/catkin/blob/kinetic-devel/cmake/test/tests.cmake#L131 as a string (`${_testing_COMMAND}`), this was our workaround to make sure spaces are not a problem.

I'm not sure why this was never hit on a ubuntu distribution; maybe it's because packages are usually installed into paths without a space by default (/etc, ...). On Windows, "Program Files" is the default location, so it's a bit easier to hit.

Reply:
I applied the quoting generically.